### PR TITLE
MAYA-110368 send ufe notifications on undo and redo

### PR DESCRIPTION
**Pull request by williamkrick**
The only thing that is not working fully consistently is when we set the value we get one Ufe::AttributeValueChanged for the height changing. On undo and redo we get two Ufe::AttributeValueChanged: one for the height and one for the bounds. I would prefer if execute, undo and redo all sent exactly the same notifications.

### CODE REVIEW

File: test/lib/mayaUsd/render/vp2RenderDelegate/testVP2RenderDelegateInteractiveWorkflows.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    """
+    Tests imaging using the Viewport 2.0 render delegate when using per-instance
+    inherited data on instances.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        # The test USD data is authored Z-up, so make sure Maya is configured
+        # that way too.
+        # cmds.upAxis(axis='z')
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/render/vp2RenderDelegate/testVP2RenderDelegateInteractiveWorkflows.py:49]: Commented out?

File: test/lib/mayaUsd/render/vp2RenderDelegate/CMakeLists.txt [hunk 2]
```diff
@@ -5,6 +5,7 @@ set(TEST_SCRIPT_FILES "")
 
 list(APPEND TEST_SCRIPT_FILES
 	testVP2RenderDelegateGeomSubset.py
+	testVP2RenderDelegateInteractiveWorkflows.py #probably ufe v2 only
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/render/vp2RenderDelegate/CMakeLists.txt:8]: The Attributes interface is UFE v2, so this needs to be moved.

**Review by ppt-adsk — CHANGES_REQUESTED** [review]

**Review by ppt-adsk — APPROVED** [review]

File: lib/mayaUsd/ufe/StagesSubject.cpp [hunk 3]
```diff
@@ -235,18 +235,18 @@ void StagesSubject::stageChanged(
             // We need to send some notifs so Maya can update (such as on undo
```
Comment (williamkrick) [inline lib/mayaUsd/ufe/StagesSubject.cpp:235]: Just above here, I could modify the loop over GetResyncedPaths to get access to the SdfChangeList::entry objects and access the same flags I get in the changedInfoOnlyPaths loop.

File: lib/mayaUsd/ufe/StagesSubject.cpp [hunk 4]
```diff
@@ -406,7 +408,20 @@ void StagesSubject::stageChanged(
...
         if (sendValueChangedFallback) {
-            valueChanged(ufePath, changedPath.GetNameToken());
+
+            // check to see if there is an entry which Ufe should notify about.
+            std::vector<const SdfChangeList::Entry*> entries = it.base()->second;
+            for (const auto& entry : entries)
+            {
+                // Adding an inert prim means we created a primSpec for an ancestor of
+                // a prim which has a real change to it.
+                if (entry->flags.didAddInertPrim || entry->flags.didRemoveInertPrim)
```
Comment (williamkrick) [inline lib/mayaUsd/ufe/StagesSubject.cpp:418]: This is the new filtering that catches the over primSpec creation & deletion. There are loads of other flags which could be relevant to check here.

**Review by ppt-adsk — APPROVED** [review]
Great work William, thanks for getting to the bottom of this!

File: lib/mayaUsd/ufe/StagesSubject.cpp [hunk 5]
```diff
@@ -235,18 +235,18 @@ void StagesSubject::stageChanged(
             // We need to send some notifs so Maya can update (such as on undo
             // to move the transform manipulator back to original position).
             const TfToken nameToken = changedPath.GetNameToken();
+            auto          usdPrimPathStr = changedPath.GetPrimPath().GetString();
+            auto ufePath = stagePath(sender) + Ufe::PathSegment(usdPrimPathStr, g_USDRtid, '/');
```
Comment (seando-adsk) [inline lib/mayaUsd/ufe/StagesSubject.cpp:239]: I have a feeling this will not pass the clang-format. Did you run it locally?

**Review by seando-adsk — APPROVED** [review]
Great work. Thanks for adding the comments to the existing attribute test.

**Review by ppt-adsk — APPROVED** [review]

**Review by seando-adsk — APPROVED** [review]

**Pull request merged**